source/common/network/filter_manager_impl.cc

```diff
@@ -73,6 +79,12 @@ FilterStatus FilterManagerImpl::onWrite() { return onWrite(nullptr, connection_)
 
 FilterStatus FilterManagerImpl::onWrite(ActiveWriteFilter* filter,
                                         WriteBufferSource& buffer_source) {
+  // Filter could return status == FilterStatus::StopIteration immediately, close the connection and
+  // use callback to call this function.
+  if (connection_.state() != Connection::State::Open) {
```

Comment:
Don't you still need the check below after calling the filter? If we are going to do these we should be complete for all cases?

Reply:
Good catch. We need to check after calling the filter.